draft-ietf-quic-applicability.md

```diff
@@ -452,12 +452,20 @@ human-readable information to an error code. Note that QUIC RESET_STREAM and
 STOP_SENDING frames provide similar capablities. Usually application error codes
 are defined to be applicabile to all three frames.
 
-Alternatively, a QUIC connection will be silently closed by each endpoint
-separately after an idle timeout. The idle timeout is announce for each
-endpoint during connection established and should be accessible by the
-application. If an application desires to keep the connection open for longer
-than the announced timeout, it can send keep-alives messages. See
-{#resumption-v-keepalive} for further guidance.
+Alternatively, a QUIC connection can be silently closed by each endpoint
+separately after an idle timeout. If enabled as indcated by a transport
+parameter in the handshake, the idle timeout is announced for each
+endpoint during connection establishment and the effective value for this
+connection is the minimum of these two advertised value. An application
+therefore should be able to configure its own maximum value as well as
+have access to the computed minimum value for this connection. An application
+may adjust the maximum idle timeout based on the number of open or expected
+connections as shorter timeout values may free-up memory more quickly.
+If an application desires to keep the connection open for longer
+than the announced timeout, it can send keep-alives messages, or a QUIC
+implementation may provide an option to defer the time-out to avoid unnesseary
```

Comment:
Not every application protocol has a keep-alive option. I view keep-alives as the responsibility of the QUIC stack instead.

Reply:
Both might have benefits depending on the application, e.g. if you anyway need to exchange certain status update messages on application layer. Should be more explicit about that in the text?